src/pyhf/optimize/opt_scipy.py

```diff
@@ -1,53 +1,56 @@
-"""scipy.optimize-based Optimizer using finite differences."""
+"""SciPy Optimizer Class."""
+from .mixins import OptimizerMixin
+import scipy
 
-from scipy.optimize import minimize
-import logging
 
-log = logging.getLogger(__name__)
+class scipy_optimizer(OptimizerMixin):
+    """
+    Optimizer that uses scipy.optimize.minimize.
+    """
 
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the scipy_optimizer.
 
-class scipy_optimizer(object):
-    """scipy.optimize-based Optimizer using finite differences."""
-
-    def __init__(self, **kwargs):
-        """Create scipy.optimize-based Optimizer."""
-        self.maxiter = kwargs.get('maxiter', 100000)
+        See :class:`pyhf.optimize.mixins.OptimizerMixin` for configuration options.
+        """
+        self.name = 'scipy'
+        super(scipy_optimizer, self).__init__(*args, **kwargs)
 
-    def minimize(
+    def _minimize(
         self,
         objective,
-        data,
-        pdf,
-        init_pars,
-        par_bounds,
+        init,
+        method='SLSQP',
+        jac=None,
+        bounds=None,
         fixed_vals=None,
-        return_fitted_val=False,
+        options={},
     ):
         """
-        Find Function Parameters that minimize the Objective.
+        Same signature as scipy.optimize.minimize.
+
+        This returns a callable that returns the fitresult.
 
         Returns:
-            bestfit parameters
-        
+            fitresult (`scipy.optimize.OptimizeResult`): the fit result
         """
+        self._minimizer = scipy.optimize.minimize
```

Comment:
if we go stateless route, maybe we don't need to assign this to `self`

Reply:
done, stateless.